### PR TITLE
add PLEC  function in spectra.py

### PR DESCRIPTION
**Pull request by davidsanchez**
Hi,

Following our email discussion, I have implemented a PLEC function in the spectra.py file

best
david

### CODE REVIEW

**Comment by cosimoNigro** [comment]
Thanks a lot @davidsanchez! Looks already good, will try to have a second deeper look later.
In the meanwhile, would you mind - if you have experience with it - adding some tests in [`agnpy/tests/test_spectra.py`](https://github.com/cosimoNigro/agnpy/blob/master/agnpy/tests/test_spectra.py)?
The failing check in the PR is related to the introduction of new code not covered by the tests.
If you are not familiar with `pytests` I'll do it in a following PR.
Thanks again!

**Comment by davidsanchez** [comment]
Hi,
I will have a look asap
thanks

File: agnpy/spectra/spectra.py [hunk 1]
```diff
@@ -179,6 +180,84 @@ def __str__(self):
...
+            gamma, self.k_e, self.p, self.gamma_c, self.gamma_min, self.gamma_max
+        )
+
+    def __str__(self):
+        return (
+            f"* electron spectrum\n"
+            + f" - power law\n"
+            + f" - k_e: {self.k_e:.2e}\n"
+            + f" - p: {self.p:.2f}\n"
+            + f" - p: {self.gamma_c:.2f}\n"
```
Comment (jsitarek) [inline agnpy/spectra/spectra.py:255]: should be gamma_c: ....

File: agnpy/spectra/spectra.py [hunk 2]
```diff
@@ -179,6 +180,84 @@ def __str__(self):
...
+        )
+
+    def __call__(self, gamma):
+        return self.evaluate(gamma, self.k_e, self.p, self.gamma_c , self.gamma_min, self.gamma_max)
+
+    @staticmethod
+    def evaluate_SSA_integrand(gamma, k_e, p, gamma_c, gamma_min, gamma_max):
+        r"""(analytical) integrand for the synchrotron self-absorption:
+        :math:`\gamma'^2 \frac{d}{d \gamma'} \left(\frac{n_e(\gamma)}{\gamma'^2}\right)`"""
+        prefactor = -(p + 2) / gamma + (-gamma / gamma_c)
```
Comment (jsitarek) [inline agnpy/spectra/spectra.py:239]: are you sure about this prefactor?
shoudn't it be 
-(p + 2) / gamma + (-1 / gamma_c)

**Comment by cosimoNigro** [comment]
Thanks for the review @jsitarek, let's see if David implements the tests, otherwise we wait for the corrections to your comments and we merge.

**Comment by davidsanchez** [comment]
I should have time to work in this during the week
best

**Comment by cosimoNigro** [comment]
No rush, thanks again.

File: agnpy/spectra/spectra.py (hunk 1 above)
Comment (davidsanchez) [reply agnpy/spectra/spectra.py]: indeed. fixed

File: agnpy/spectra/spectra.py (hunk 2 above)
Comment (davidsanchez) [reply agnpy/spectra/spectra.py]: I have fixed that

Comment (jsitarek) [reply agnpy/spectra/spectra.py]: great, thanks

**Comment by cosimoNigro** [comment]
Had a second look and looks good to me, thanks a lot @davidsanchez and welcome among the agnpy developers!

**Pull request merged**